### PR TITLE
Inertia and Subscription Validation for alerts

### PR DESCRIPTION
**Pull request by studanshu**

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java [hunk 1]
```diff
@@ -480,5 +490,41 @@ public String value() {
         }
 
     }
+
+    private Long getMaximumIntervalLength(String expression) {
+        expression = "@" + expression.replaceAll("[\\s\\t\\r\\n\\f]*", "");
```
Comment (sundeepsf) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java]: seems like this extracting logic is very similar to the logic for computing absolute expressions. Maybe you can move the common logic to some kind of util class?

Comment (studanshu) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java]: Yes. I also thought so. But regexes are a bit different in both the cases.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Notification.java [hunk 2]
```diff
@@ -431,6 +434,16 @@ public void setNotifierName(String notifierName) {
      */
     public void setSubscriptions(List<String> subscriptions) {
         this.subscriptions.clear();
+
+        for(String currentSubscription: subscriptions) {
+            if (this.getNotifierName().equals(AlertService.SupportedNotifier.GUS.getName())) {
+                if (currentSubscription.length() < 10)
+                    throw new IllegalArgumentException("GUS subjectId is incorrect.");
```
[the remark this answers is not in the document]
Comment (studanshu) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Notification.java]: https://github.com/salesforce/Argus/blob/develop/ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/notifier/GusNotifier.java#L199
We specify subjectId over here.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java (hunk 1 above)
Comment (sundeepsf) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java]: Seems this logic will be useful in other places also. This function seems out of place in trigger class. Could you move it to a util class?

Comment (studanshu) [reply ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Trigger.java]: Ok.

**Pull request merged**